src/hooks/safe-apps/useSafeAppFromBackend.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react'
+import { getSafeApps, SafeAppData } from '@gnosis.pm/safe-react-gateway-sdk'
+import { Errors, logError } from '@/services/exceptions'
+import useAsync, { AsyncResult } from '../useAsync'
+import { trimTrailingSlash } from '@/utils/url'
+
+const useSafeAppFromBackend = (url: string, chainId: string): AsyncResult<SafeAppData> => {
+  const [backendApp, error, loading] = useAsync(async () => {
+    // We do not have a single standard for storing URLs, it may be stored with or without a trailing slash.
+    // But for the request it has to be an exact match.
+    const retryUrl = url.endsWith('/') ? trimTrailingSlash(url) : `${url}/`
+    let response = await getSafeApps(chainId, { url })
+    if (!response?.[0]) {
```

Comment:
The return type for `response` suggests that it can't be undefined so we should be able to do `response[0]`.

Also for the future, I think we should consider handling this logic in the gateway so that we only have to do one request. Or alternatively, standardize all safe apps urls.

Reply:
```
export type AsyncResult<T> = [result: T | undefined, error: Error | undefined, loading: boolean]
```

The undefined comes from the hook

Reply:
> The undefined comes from the hook

I meant that its possible to do `response[0]` instead of `response?.[0]` since `response` itself can't be undefined.

Reply:
Ah, you're right. Fixed it now